### PR TITLE
For #6804: Adds private browsing label to add-on

### PR DESCRIPTION
**Pull request by ValentinTimisica**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks.
- [x] **Tests**: This PR does not include thorough tests.
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md).
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt [hunk 1]
```diff
@@ -327,7 +333,9 @@ class AddonsManagerAdapter(
         val addonSummaryTextColor: Int? = null,
         val sectionsTypeFace: Typeface? = null,
         @ColorRes
-        val addonBackgroundIconColor: Int? = null
+        val addonBackgroundIconColor: Int? = null,
+        @DrawableRes
+        val addonAllowPrivateBrowsingLabelDrawableRes: Int? = null
```
Comment (Amejia481) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt:338]: Should we also expose a colour to change the tint of the icon, as in Fenix icons tint can change depending on the theme colour ( dark, light)? I'm not sure if this icon should adapt to theme changes

**Comment by ValentinTimisica** [comment]
@psymoon the design was proposed by UX team [here](https://github.com/mozilla-mobile/fenix/issues/10244#issue-607628526) and I followed it in my patch. Regarding the UX feedback I left a [comment](https://github.com/mozilla-mobile/fenix/issues/10244#issuecomment-620671858) on the task and I'm waiting for the response, but if I have to make changes to this feature, it will most likely have to be made on Fenix ([here is the PR](https://github.com/mozilla-mobile/fenix/pull/10317)).

On the subject of squashing into one commit my opinion is that we should keep multiple small commits. In my experience I saw that breaking the work down has multiple advantages both for the review phase, where the work can be reviewed in smaller chunks and for the maintenance phase where a past change has clearer motives and is tied to fewer changes although for the ticket there were modifications needed in 10+ files.

I prefer multiple commits and I see them as necessary for basically the same reasons listed in [Google's commits best practices](https://github.com/google/eng-practices/blob/master/review/developer/small-cls.md)

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt (hunk 1 above)
Comment (ValentinTimisica) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt:338]: Hello @Amejia481, I exposed directly the icon (addonAllowPrivateBrowsingLabelDrawableRes) with the colors changed depending on the theme ([here](https://github.com/mozilla-mobile/fenix/blob/b3b0fdf1e13ca9dcebe730b44aab5290cff40a02/app/src/main/res/drawable/ic_add_on_private_browsing_label.xml#L1) is the svg from Fenix where I used addOnPrivateBrowsingExteriorCircleBackground and addOnPrivateBrowsingInteriorIconBackground attributes to change the colors depending on the theme. After that, I pass this resource in AddonsManagerAdapter.Style constructor. You can check it [here](https://github.com/mozilla-mobile/fenix/blob/b3b0fdf1e13ca9dcebe730b44aab5290cff40a02/app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt#L131).)

Comment (Amejia481) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt:338]: If colour is changes when switching themes it's OK  :)

**Comment by psymoon** [comment]
bors r=Amejia481,psymoon

**Pull request merged**